packages/sdk38/src/lcdapi/staking.spec.ts

```diff
@@ -0,0 +1,508 @@
+import { Secp256k1Wallet } from "../wallet";
+import { LcdClient } from "./lcdclient";
+import { setupStakingExtension, StakingExtension } from "./staking";
+
+function makeStakingClient(apiUrl: string): LcdClient & StakingExtension {
+  return LcdClient.withExtensions({ apiUrl }, setupStakingExtension);
+}
+
+describe("StakingExtension", () => {
+  const defaultFee = {
+    amount: coins(25000, "ucosm"),
+    gas: "1500000", // 1.5 million
+  };
+
+  beforeAll(async () => {
+    if (wasmdEnabled()) {
+      const wallet = await Secp256k1Wallet.fromMnemonic(faucet.mnemonic);
+      const client = new SigningCosmosClient(wasmd.endpoint, faucet.address, wallet, {});
+
+      const chainId = await client.getChainId();
+      {
+        const msg: MsgDelegate = {
+          type: "cosmos-sdk/MsgDelegate",
+          value: {
+            delegator_address: faucet.address,
+            validator_address: validatorAddress,
+            amount: coin(25000, "ustake"),
+          },
+        };
+        const memo = "Test delegation for wasmd";
+        const { accountNumber, sequence } = await client.getNonce();
+        const signBytes = makeSignBytes([msg], defaultFee, chainId, memo, accountNumber, sequence);
+        const signature = await wallet.sign(faucet.address, signBytes);
+        const tx = {
+          msg: [msg],
+          fee: defaultFee,
+          memo: memo,
+          signatures: [signature],
+        };
+
+        const receipt = await client.postTx(tx);
+        assert(!isPostTxFailure(receipt));
+      }
+      {
+        const msg: MsgUndelegate = {
+          type: "cosmos-sdk/MsgUndelegate",
+          value: {
+            delegator_address: faucet.address,
+            validator_address: validatorAddress,
+            amount: coin(100, "ustake"),
+          },
+        };
+        const memo = "Test undelegation for wasmd";
+        const { accountNumber, sequence } = await client.getNonce();
+        const signBytes = makeSignBytes([msg], defaultFee, chainId, memo, accountNumber, sequence);
+        const signature = await wallet.sign(faucet.address, signBytes);
+        const tx = {
+          msg: [msg],
+          fee: defaultFee,
+          memo: memo,
+          signatures: [signature],
+        };
+
+        const receipt = await client.postTx(tx);
+        assert(!isPostTxFailure(receipt));
+      }
+
+      await sleep(75); // wait until transactions are indexed
+    }
+  });
+
+  describe("delegatorDelegations", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeStakingClient(wasmd.endpoint);
+      const response = await client.staking.delegatorDelegations(faucet.address);
+      expect(response).toEqual({
+        height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        result: [
+          {
+            delegator_address: "cosmos1pkptre7fdkl6gfrzlesjjvhxhlc3r4gmmk8rs6",
+            validator_address: "cosmosvaloper1gjvanqxc774u6ed9thj4gpn9gj5zus5u32enqn",
```

Comment:
We can use this from `validatorAddress` in `packages/sdk38/src/testutils.spec.ts`